### ISSUE TITLE
[Question] typescript import enum

### ISSUE THREAD

**Issue by hyusetiawan** [microsoft/playwright#13265]
I am using next.js / react and inside the page.tsx I export an enum called `TestID`, I use it to mark specific component with test id that I will then share the same test id in jest. I want to do the same thing with playwright but it seems I am getting `undefined` as the enum, as a matter of fact, during runtime `TestID`is a type of `string` how come? Is there a way to share the test ids this way? Trying to avoid hard coding.

**Comment by rwoll** [comment]
Let me make sure I understand the setup. Is this what your Playwright Test import structure/file layout looks like?

```ts
// @filename example.spec.ts
import { MyEnum } from './component'; // THIS IS A TSX FILE!

test('works', () => {
  expect(MyEnum.SomeID).toBe("value");
});
```

```ts
// @filename component.tsx
export enum MyEnum {
    SomeID = "value";
}
```

If so, I repro'd the behavior: https://github.com/microsoft/playwright/pull/13272. For now, as a work around, you can move the enums into a .ts file (`shared.ts`) and import it into your test as well as your component.

**Comment by hyusetiawan** [comment]
@rwoll that setup is correct, glad you are able to reproduce it, is the issue caused because it's tsx?

**Comment by rwoll** [comment]
> is the issue caused because it's tsx

Yes—I received confirmation: "[w]e don't compile TSX, we only stub those altogether for component testing. So this test documented present behavior." [^1]

If you move your enum's into a `.ts` file and import that, you should be good. For now, avoid import'ing `.tsx` files into your tests. They won't do what you expect. (We are experimenting with component testing, so at some point in the future, there may be legitimate reasons to import a `.tsx` file into your test, but not atm.)

[^1]: https://github.com/microsoft/playwright/pull/13272#discussion_r841322948
